components/CollectionContainer/CollectionContainer.js

```diff
@@ -20,20 +20,27 @@ export default function CollectionContainer({
   color,
   cards,
   deleteCollection,
-  isMastered,
+  archive = false,
+  resetCard,
 }) {
+  function handleResetCards() {
+    cards.forEach(async (card) => {
+      await resetCard(card);
+    });
+  }
```

Comment:
**Please check if your function  behave s as you expect.** 

Using `async/await` inside a `.forEach()` loop isn't "wong," but it may not behave as you might expect due to how `.forEach()` handles asynchronous operations.

💡 The `.forEach()` loop [doesn't wait for promises](https://medium.com/illumination/javascripts-foreach-and-async-code-with-great-power-comes-well-you-know-21c1b489576f) to resolve within its callback function. This means if you have asynchronous operations inside .forEach(), those operations will be initiated, but the loop won't wait for each operation to complete before moving on to the next iteration. As a result, all asynchronous operations are essentially started at the same time, similar to firing off multiple parallel asynchronous tasks without waiting for any to complete before starting the next. This can lead to unexpected behaviour.

Reply:
jes, we already discussed this issue, but decide to go with this. In collections/[id] line 71/72 we point this out for code reviewers:   //for each starts an asynchronous call for each card, but does not await the outcomes.
      //This is ok here, because we don't do anything here afterwards anyways.

Reply:
why do you need the async here anyway?

Reply:
To make it explicit that `resetCard()` is returning a `Promise` and we are aware of this.  It's indeed not needed here **right now**, since we are not awaiting the call.  But it helps future developers and the team to extend this code file later by making this fact explicit and thus communicating to them that they must be careful here. The additional await does not harm (I think?)?